test/test_mixed_null_assertions.cpp

```diff
@@ -78,3 +78,206 @@ TEST(List_Mixed_do_insert)
+        CHECK(list.size() == 1);
+    }
+}
+
+TEST(Mixed_Set_unresolved_as_null)
+{
+    Group g;
+
+    auto t = g.add_table("foo");
+    t->add_column_set(type_Mixed, "mixeds");
+    auto obj = t->create_object();
+    auto obj1 = t->create_object();
+    auto obj2 = t->create_object();
+    auto set = obj.get_set<Mixed>("mixeds");
+    auto [it, success] = set.insert(Mixed{obj1});
+    obj1.invalidate();
+
+    CHECK(success);
+    auto [it1, success1] = set.insert(Mixed{"test"});
+    CHECK(success1);
+
+    {
+        // null treated as invalalid link
+        CHECK(set.size() == 2);
+        auto [it, success] = set.insert(Mixed{});
+        CHECK(!success);
+        auto [it1, success1] = set.insert_null();
+        CHECK(!success1);
+        CHECK(set.size() == 2);
+    }
+
+    {
+        // find all mixed nulls or unresolved link should work the same way
+        // there can only be 1 NULL
+        int cnt = 0;
+        set.find_all(realm::null(), [this, &set, &cnt](size_t pos) {
+            CHECK(pos != not_found);
+            CHECK(set.is_null(pos));
+            cnt += 1;
+        });
+        CHECK(cnt == 1);
+    }
+
+    {
+        auto index = set.find_any(realm::null());
+        CHECK(index != not_found);
+        CHECK(set.is_null(index));
+    }
+
+    {
+        auto [it, success] = set.insert(Mixed{obj2});
+        obj2.invalidate();
+        CHECK(success);
+        CHECK(set.size() == 3);
+        std::vector<size_t> indices{1, 0, 2};
+        set.sort(indices);
+        CHECK(indices.size() == 3);
+        CHECK(indices[0] == 0);
+        CHECK(indices[1] == 1);
+        CHECK(indices[2] == 2);
+    }
+
+    {
+        // trigger interface exception, we ended up with multiple nulls
+        Group g;
+        auto t = g.add_table("foo");
+        t->add_column_set(type_Mixed, "mixeds");
+        auto obj = t->create_object();
+        auto obj1 = t->create_object();
+        auto obj2 = t->create_object();
+        auto set = obj.get_set<Mixed>("mixeds");
+        auto [it, success] = set.insert(obj1);
+        auto [it1, success1] = set.insert(obj2);
+        CHECK(success);
+        CHECK(success1);
+        CHECK(set.size() == 2);
+        CHECK(!set.is_null(0));
+        CHECK(!set.is_null(1));
+        obj1.invalidate();
+        CHECK(set.is_null(0));
+        CHECK(!set.is_null(1));
+        obj2.invalidate();
+        // this is the only violation we allow right now, we have ended up with 2 nulls
+        CHECK(set.is_null(0));
+        CHECK(set.is_null(1));
+        auto obj3 = t->create_object();
+        set.insert(obj3);
+        CHECK(set.size() == 3);
+        // we can now do find_all for nulls (only actual null values will be returned, no unresolved links)
+        set.find_all(Mixed{}, [this, &set](size_t index) {
+            CHECK(index == 0);
+            CHECK(set.is_null(index));
+        });
+        // erase null will erase only nulls (no unresolved)
```

Comment:
I don't understand this comment. Will the now unresolved entries not be removed?

Reply:
No, the agreement was that unresolved links should not be removed, only real null values are....

Reply:
This because we don't want to break sync.